dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionHelper.java

```diff
@@ -87,18 +87,24 @@ public void evictAndAddEntries(Collection collection) {
             entry.setLabels(labelStrings);
             List<CategorySummary> summaries = entryDAO.findCategorySummariesByEntryId(entry.getId());
             entry.setCategorySummaries(summaries);
-            if (entry.getEntryType().equals("tool")) {
+            switch (entry.getEntryType()) {
+            case "tool":
                 entry.setDescriptorTypes(entryDAO.getToolsDescriptorTypes(entry.getId()));
-            } else if (entry.getEntryType().equals("workflow")) {
+                break;
+            case "workflow":
                 entry.setDescriptorTypes(entryDAO.getWorkflowsDescriptorTypes(entry.getId()));
+                break;
+            case "apptool":
+                entry.setDescriptorTypes(entryDAO.getWorkflowsDescriptorTypes(entry.getId()));
+                // we get file descriptor types like workflows, but make the UI treat these as workflows (so icon and url work)
+                entry.setEntryType("tool");
```

Comment:
Given my experience thus far with SEAB-4040, I'm a little concerned about changing the apptool entry type to "tool" here.  In SEAB-4040 at least, when the UI lists the tools, the logic to render an apptool is different than a regular tool, so the UI needs to know the actual type...

Reply:
Might be a different component? 
For now, I hit this logic 
https://github.com/dockstore/dockstore-ui2/blob/2.9.0-beta.0/src%2Fapp%2Forganizations%2Fcollection%2Fcollection.component.html#L105-L113
which only has the two choices

Reply:
Looks like this UI portion only calls on Entry level properties so seems fine. I guess to be paranoid, you could run the cypress test against this version of the web service. The GitHub app tools testing isn't super extensive, but it's something i guess.

Reply:
Will do, its on the checklist anyway